frontend/src/metabase/visualizations/visualizations/SmartScalar/utils.ts

```diff
@@ -63,3 +79,244 @@ export const getValueHeight = (height: number): number => {
 export const getChangeWidth = (width: number): number => {
   return Math.max(width - ICON_SIZE - ICON_MARGIN_RIGHT - 2 * SPACING, 0);
 };
+
+export const COMPARISON_TYPES = {
+  PREVIOUS_VALUE: "previousValue",
+  PREVIOUS_PERIOD: "previousPeriod",
+  PERIODS_AGO: "periodsAgo",
+} as const;
+
+export type ComparisonTypes =
+  typeof COMPARISON_TYPES[keyof typeof COMPARISON_TYPES];
+
+export const COMPARISON_SELECTOR_OPTIONS = {
```

Comment:
@kulyk You'll want to add the new selector option here with the various information it needs 

```javascript
CUSTOM_VALUE: {
  type: COMPARISON_TYPES.CUSTOM_VALUE,
  name: t`...`,
  MenuItemComponent: CustomValueSelectorWidget
}
```